src/api/v2/v2AzureResourcesApi.ts

```diff
@@ -44,6 +44,8 @@ export interface ApplicationResource extends ResourceBase {
     readonly tags?: {
         [propertyName: string]: string;
     };
+    /** A copy of the raw resource intended for the viewProperties command */
+    _raw: Record<string, unknown>;
```

Comment:
A couple of thoughts.

1. Can we rename to `raw`? This doesn't need to be regarded as a private property.
2. Will extensions want this View Properties command to show up for subtree items they own? If so, we should move this field to `ResourceBase`.

Reply:
> Can we rename to `raw`? This doesn't need to be regarded as a private property.

Sure

> Will extensions want this View Properties command to show up for subtree items they own? If so, we should move this field to ResourceBase.

I thought about this but wasn't sure. Sounds like a good idea and we could remove some code in the client extensions. However, some client extensions have async view properties commands that fetch additional data before showing the properties. So not sure if we'd want to try to handle that vs. just letting the clients control all items they own.

Reply:
For deeply custom behavior I think they can just DIY it. I bet it's sufficient most of the time to kick off loading that data and getting it attached to `.raw`, ignoring the small chance of the user right click -> View Properties before the load is finished.